snapcraft/plugins/catkin.py

```diff
@@ -803,13 +842,9 @@ def _build_catkin_packages(self):
         catkincmd.append("-DCMAKE_BUILD_TYPE={}".format(build_type))
 
         # Finally, add any cmake-args requested from the plugin options
-        catkincmd.extend(self.options.catkin_cmake_args)
+        catkincmd.extend(self._parse_cmake_args())
 
-        # This command must run in bash due to a bug in Catkin that causes it
-        # to explode if there are spaces in the cmake args (which there are).
-        # This has been fixed in Catkin Tools... perhaps we should be using
-        # that instead.
```

Comment:
It's been a long time since this was written. Any chance you tried just switching this to `self.run` to observe the breakage for yourself? I don't remember what exactly it was.

Reply:
I did, and at first thought it was OK.  but then I came up with a case that breaks:

-DCMAKE_C_FLAGS="-Wall -Werror -DIGNORE_ME_IM_JUST_TESTING_SPACES"

cmake /root/parts/ros-project/build/src/listener -DCATKIN_DEVEL_PREFIX=/root/parts/ros-project/build/devel_isolated/listener -DCMAKE_INSTALL_PREFIX=/root/parts/ros-project/install/opt/ros/melodic -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS="-Wall -Werror -DIGNORE_ME_IM_JUST_TESTING_SPACES" -G Unix Makefiles in '/root/parts/ros-project/build/build_isolated/listener'

/usr/bin/cc   "-Wall -Werror -DIGNORE_ME_IM_JUST_TESTING_SPACES"    -o CMakeFiles/cmTC_64abc.dir/testCCompiler.c.o   -c /root/parts/ros-project/build/build_isolated/listener/CMakeFiles/CMakeTmp/testCCompiler.c

Reply:
Ah excellent, okay. And I see you've used essentially that for the spread test as well, so this seems to work just fine.